src/file-upload/dropzone.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './styles.css.js';
+import clsx from 'clsx';
+import InternalIcon from '../icon/internal';
+import InternalBox from '../box/internal';
+
+interface DropzoneProps {
+  onChange: (files: File[]) => void;
+  children: React.ReactNode;
+}
+
+export function useDropzoneVisible() {
+  const [isDropzoneVisible, setDropzoneVisible] = useState(false);
+
+  // Registering global drag events listeners.
+  useEffect(() => {
+    // The timer helps avoiding dropzone blinking.
+    let dragTimer: null | ReturnType<typeof setTimeout> = null;
+
+    // The file-upload dropzone is shown when the user drags files over the to the browser.
+    const onDocumentDragOver = (event: DragEvent) => {
+      event.preventDefault();
```

Comment:
Why is this line needed, what is the default behaviour? If multiple dropzones are on a page, does this prevent the others from working? Should we move this line inside the conditional block below?

Reply:
This prevents browser's standard treatment, see example: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dragover_event

This line does not prevent the event from being received by the dropzones.

That said, there might be a case where the global listener that we register does cause trouble for the pages that do expect some default behaviour of the dragged files. Once reported we can think of moving the line somewhere or adding a flag. For example, react-dropzone has a flag to allow document drop on the page: https://github.com/react-dropzone/react-dropzone/blob/master/src/index.js#L64.